### ISSUE TITLE
No file for valminusminival for dataset year 2017. How is valminusminival obtained?

### ISSUE THREAD

**Issue by JonathanCMitchell** [matterport/Mask_RCNN#376]
I am attempting to train on 2017 coco data. There is no valminusminival2017.json file. 

The instances_valminusminival json files are for 2014 data only. There isn't one for 2017 data. I am guessing you can rename file `instances_valminusminival2014.json` to `instances_valminusminival2017.json` but I am not sure how this will affect your validation loss. 

Does anybody know what the instances_valminusminival dataset consists of ? / How it was computed? I simply commented out [L477](https://github.com/matterport/Mask_RCNN/blob/master/coco.py#L477) in order to train but I don't know if I am missing something.

**Comment by fastlater** [comment]
@JonathanCMitchell  Before rename the file, better go to http://cocodataset.org/#download and download the 2017 data. The json file contains the dataset info, images info and segmentation info of each specific image. Actually, I guess that if you rename the file, I don't think the script will notice that but you will be training using the 2014 dataset, not the 2017 dataset.

**Comment by JonathanCMitchell** [comment]
Hi @fastlater. I see the validation data, but there isn't anything called valminusminival2017. Are most people simply training on the 2014 dataset? If my train is `train2017` and my val has been renamed from 2014->2017 then would that mean I am training on the 2014 dataset even though my train folder is 2017?

**Comment by hochthom** [comment]
Simply replace the following lines (478-488) with:

        # Training dataset. Use the training set and 35K from the
        # validation set, as as in the Mask RCNN paper.
        dataset_train = CocoDataset()
        dataset_train.load_coco(args.dataset, "train", year=args.year, auto_download=args.download)
        dataset_train.prepare()

        # Validation dataset
        dataset_val = CocoDataset()
        dataset_val.load_coco(args.dataset, "val", year=args.year, auto_download=args.download)
        dataset_val.prepare()
